spotipy/oauth2.py

```diff
@@ -374,9 +374,17 @@ def _open_auth_url(self):
         except webbrowser.Error:
             logger.error("Please navigate here: %s", auth_url)
 
-    def _get_auth_response_interactive(self):
-        self._open_auth_url()
-        response = SpotifyOAuth._get_user_input("Enter the URL you were redirected to: ")
+    def _get_auth_response_interactive(self, open_browser=True):
+        if open_browser:
+            self._open_auth_url()
+            prompt = "Enter the URL you were redirected to: "
+        else:
+            url = self.get_authorize_url()
+            prompt = (
+                "Go to the following URL: {}\n"
+                "Enter the URL you were redirected to: ".format(url)
+            )
```

Comment:
Why this condition? Aren't both sides doing the same thing?

Reply:
In the case where we want to open the browser, the method `_open_auth_url` will launch the browser with the authorization URL. Otherwise, this authorization URL is printed out to stdout and the user needs to figure out how to open it by himself. In both cases we ask the user for the final URL she has been redirected to.